File: frontend/src/types/support.ts
export interface SupportMessage {
  uuid: string
  user_id: string
  user_name: string | null
  content: string
  is_support_reply: boolean
  created_at: string | null
}

export interface SupportAttachment {
  uuid: string
  filename: string
  file_type: string | null
  uploaded_by: string
  message_uuid: string | null
  created_at: string | null
}

export interface SupportTicket {
  uuid: string
  subject: string
  status: 'open' | 'in_progress' | 'closed'
  priority: 'low' | 'normal' | 'high'
  user_id: string
  user_name: string | null
  user_email: string | null
  team_id: string | null
  assigned_to: string | null
  messages: SupportMessage[]
  attachments: SupportAttachment[]
  message_count: number
  created_at: string | null
  updated_at: string | null
  closed_at: string | null
}

export interface SupportTicketSummary {
  uuid: string
  subject: string
  status: 'open' | 'in_progress' | 'closed'
  priority: 'low' | 'normal' | 'high'
  user_id: string
  user_name: string | null
  assigned_to: string | null
  message_count: number
  last_message_preview: string | null
  last_message_at: string | null
  created_at: string | null
  updated_at: string | null
  closed_at: string | null
}

export interface SupportContact {
  user_id: string
  email: string
  name: string
}

File: backend/app/models/support.py
"""Support ticket models."""

import datetime
import uuid as uuid_mod
from enum import Enum
from typing import Optional

from beanie import Document
from pydantic import BaseModel, Field


class TicketStatus(str, Enum):
    OPEN = "open"
    IN_PROGRESS = "in_progress"
    CLOSED = "closed"


class TicketPriority(str, Enum):
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"


class SupportMessage(BaseModel):
    """Embedded message within a ticket conversation."""

    uuid: str = Field(default_factory=lambda: uuid_mod.uuid4().hex)
    user_id: str
    user_name: Optional[str] = None
    content: str
    is_support_reply: bool = False
    created_at: datetime.datetime = Field(
        default_factory=lambda: datetime.datetime.now(datetime.timezone.utc)
    )


class SupportAttachment(BaseModel):
    """File attached to a support ticket."""

    uuid: str = Field(default_factory=lambda: uuid_mod.uuid4().hex)
    filename: str
    file_type: Optional[str] = None
    file_data: str = ""  # base64 encoded
    uploaded_by: str
    message_uuid: Optional[str] = None  # linked to a specific message
    created_at: datetime.datetime = Field(
        default_factory=lambda: datetime.datetime.now(datetime.timezone.utc)
    )


class SupportTicket(Document):
    uuid: str = Field(default_factory=lambda: uuid_mod.uuid4().hex)
    subject: str
    status: TicketStatus = TicketStatus.OPEN
    priority: TicketPriority = TicketPriority.NORMAL

    # Creator
    user_id: str
    user_name: Optional[str] = None
    user_email: Optional[str] = None
    team_id: Optional[str] = None

    # Conversation
    messages: list[SupportMessage] = []
    attachments: list[SupportAttachment] = []

    # Assignment
    assigned_to: Optional[str] = None  # user_id of support person

    # Timestamps
    created_at: datetime.datetime = Field(
        default_factory=lambda: datetime.datetime.now(datetime.timezone.utc)
    )
    updated_at: datetime.datetime = Field(
        default_factory=lambda: datetime.datetime.now(datetime.timezone.utc)
    )
    closed_at: Optional[datetime.datetime] = None

    class Settings:
        name = "support_ticket"
        indexes = [
            "uuid",
            "user_id",
            "status",
            "assigned_to",
            [("status", 1), ("created_at", -1)],
            [("user_id", 1), ("created_at", -1)],
        ]
